zend_abstract_interface/sandbox/php8/sandbox.c

```diff
@@ -4,7 +4,30 @@ extern inline void zai_sandbox_open(zai_sandbox *sandbox);
 extern inline void zai_sandbox_close(zai_sandbox *sandbox);
 
 extern inline void zai_sandbox_error_state_backup(zai_error_state *es);
-extern inline void zai_sandbox_error_state_restore(zai_error_state *es);
+
+/* In PHP 8, we cannot extern inline zai_sandbox_error_state_restore() because
+ * zend_string_release() is static which generates the following compiler
+ * warning:
+ *
+ *   "warning: 'zend_string_release' is static but used in inline function
+ *    'zai_sandbox_error_state_restore' which is not static"
+ */
```

Comment:
Just to document some history: I tried talking Nikita into allowing me to fix these in php-src but he didn't see the value in doing `inline` + `extern inline` vs allowing the compiler to do it via `static inline`. Plus there are `zend_always_inline` cases which have _theoretically_ been put there because of performance reasons, and we wouldn't want them to accidentally _not_ get inlined (although "always" is technically a lie anyway). I didn't feel like doing the necessary performance A/B work to demonstrate there weren't regressions and dropped it.

tl;dr It'll probably never get fixed upstream.